core/src/test/java/io/fabric8/maven/core/util/MergeResourceTest.java

```diff
@@ -0,0 +1,236 @@
+/*
```

Comment:
Nice coverage!
May be we can improve the test coverage for following inputs
- Metadata for resource(main resource) is absent
- DeploymentSpec for resource is absent
- PodTemplateSpec for resource is absent
- PodSpec for resource is absent
- Expected behavior if switchOnLocalCustomisation is true at couple of instance